File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.py
class Solution:
    
    def countGoodSubstrings(self, s: str) -> int:
        
        if len(s) < 3:
            return 0
        def default_zero():
            return 0
        
        freq = defaultdict(default_zero)
        dupcount = 0
        result = 0
        
        for i in range(3):
            freq[s[i]]+=1
            if freq[s[i]] == 2:
                dupcount+=1
        
        if dupcount == 0:
            result += 1
        
        for i in range(3,len(s)):
            freq[s[i]] += 1
            if freq[s[i]] == 2:
                dupcount+=1
            
            freq[s[i-3]] -= 1
            if freq[s[i-3]] == 1:
                dupcount-=1
            
            if dupcount == 0:
                result+=1
        
        return result
            
            
            
            
            
        

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        int count = 0;
        for(int i=0; i<s.length()-2; i++){
            char a = s.charAt(i);
            char b = s.charAt(i+1);
            char c = s.charAt(i+2);
            if((a!=b) && (b!=c) && (a!=c)) count++;
        }
        
        return count;
    }
}


/*


owuxoelszb
  i
j

o - 1
w - 1
u - 1
*/